src/main/java/org/jitsi/jicofo/jigasi/JigasiDetector.java

```diff
@@ -91,45 +111,121 @@ protected void onInstanceStatusChanged(
     protected void notifyInstanceOffline(Jid jid)
     {}
 
+    /**
+     * Selects the jigasi instance that is less loaded.
+     * @param preferredRegions a list of preferred regions.
+     * @return XMPP address of Jigasi instance or <tt>null</tt> if there are
+     * no Jigasis available currently.
+     */
+    public Jid selectTranscriber(
+        List<Jid> filter, List<String> preferredRegions)
+    {
+        return this.selectJigasi(
+            instances, filter, preferredRegions, localRegion, true);
+    }
+
     /**
      * Selects the jigasi instance that is less loaded.
      *
+     * @param filter a list of <tt>Jid</tt>s to be filtered from the list of
+     * available Jigasi instances. List that we do not want as a result.
+     * @param preferredRegions a list of preferred regions.
      * @return XMPP address of Jigasi instance or <tt>null</tt> if there are
      * no Jigasis available currently.
      */
-    public Jid selectJigasi()
+    public Jid selectJigasi(List<Jid> filter, List<String> preferredRegions)
     {
-        return this.selectJigasi(null);
+        return selectJigasi(
+            instances, filter, preferredRegions, localRegion, false);
     }
 
     /**
      * Selects the jigasi instance that is less loaded.
      *
      * @param filter a list of <tt>Jid</tt>s to be filtered from the list of
      * available Jigasi instances. List that we do not want as a result.
+     * @param preferredRegions a list of preferred regions.
+     * @param transcriber Whether we need to select a transcriber or sipgw
+     * jigasi instance.
      * @return XMPP address of Jigasi instance or <tt>null</tt> if there are
      * no Jigasis available currently.
      */
-    public Jid selectJigasi(List<Jid> filter)
+    public static Jid selectJigasi(
+        List<BrewInstance> instances,
+        List<Jid> filter,
+        List<String> preferredRegions,
+        String localRegion,
+        boolean transcriber)
     {
-        BrewInstance lessLoadedInstance = null;
-        int numberOfParticipants = Integer.MAX_VALUE;
-        for (BrewInstance jigasi : instances)
+        // let's filter using the provided filter and those in graceful shutdown
+        List<BrewInstance> filteredInstances = instances.stream()
+            .filter(j -> filter == null || !filter.contains(j.jid))
+            .filter(j -> j.status == null
+                || !Boolean.parseBoolean(j.status.getValueAsString(
```

Comment:
Can we add a method to `BrewInstance` which makes this check? For example `isInGracefulShutdown`?